WordPress/Classes/ViewRelated/Activity/ActivityListViewModel.swift

```diff
@@ -40,17 +40,27 @@ enum ActivityListViewModel {
         case .loading, .error:
             return .Empty
         case .ready(let activities):
-            let rows = activities.map({ activity in
+            let rows = activities.filter({ activity in
+                !activity.isDiscarded
+            }).map({ activity in
                 return ActivityListRow(
                     activity: activity,
                     action: { (row) in
                         presenter.presentRewindFor(activity: activity)
                     }
                 )
             })
-            return ImmuTable(sections: [
-                ImmuTableSection(rows: rows)
-            ])
+
+            let publishedDateUTCKeyPath = \ActivityListRow.activity.publishedDateUTCWithoutTime
+            let groupedRows = rows.sortedGroup(keyPath: publishedDateUTCKeyPath)
+
+            let tableSections = groupedRows.map { (date, rows) in
+                return ImmuTableSection(headerText: date,
+                                        optionalRows: rows,
+                                        footerText: nil)
+            }
+
+            return ImmuTable(optionalSections: tableSections)
```

Comment:
I don't think you'd get any `nil` rows coming from `sortedGroup`, but the compiler doesn't know that, so I guess the `optional` ImmuTable variants are 👌 here